docs/development/plugin/development-elasticsearch.asciidoc

```diff
@@ -0,0 +1,41 @@
+[[development-elasticsearch]]
+=== Communicating with elasticsearch
+
+Kibana exposes two clients on the server and browser for communicating with elasticsearch.
+There is an 'admin' client which is used for managing Kibana's state, and a 'data' client for all
+other requests.  The clients use the elasticsearch.js library.
```

Comment:
Plugin developers should actually not be using the elasticsearch-js library directly. For browsers they are using Courier, and the server they should be using `callWithInternalUser` or `callWithRequest`